engine/src/main/java/org/terasology/engine/world/chunks/remoteChunkProvider/ReceivedChunkInitialProvider.java

```diff
@@ -0,0 +1,35 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.engine.world.chunks.remoteChunkProvider;
+
+import org.joml.Vector3ic;
+import org.terasology.engine.world.chunks.Chunk;
+import org.terasology.engine.world.chunks.pipeline.InitialChunkProvider;
+
+import java.util.Comparator;
+import java.util.Set;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.PriorityBlockingQueue;
+
+public class ReceivedChunkInitialProvider implements InitialChunkProvider {
+    private BlockingQueue<Chunk> chunkQueue;
+
+    public ReceivedChunkInitialProvider(Comparator<Chunk> comparator) {
+        chunkQueue = new PriorityBlockingQueue<>(800, comparator);
+    }
+
+    public void submit(Chunk chunk) {
+        chunkQueue.add(chunk);
+    }
+
+    @Override
+    public boolean hasNext() {
+        return !chunkQueue.isEmpty();
+    }
+
+    @Override
+    public synchronized Chunk next(Set<Vector3ic> currentlyGenerating) {
+        return chunkQueue.poll();
```

Comment:
Here we don't make use of `currentlyGenerating` - is this on purpose?

Reply:
As far as I can tell, that parameter is the only thing thing that distinguishes this class from the `Queue<Chunk>` that is backing it.

Reply:
oh, I see the _other_ implementation of this interface does make use of it, though this does not.

Reply:
In general, the client doesn't receive duplicate chunks, so it doesn't need to check; and if it did receive the same chunk from the server again before it's done processing, the new one is more up-to-date, so it should keep the new one.